sql/moz-fx-data-shared-prod/mozilla_vpn_derived/waitlist_v1/query.sql

```diff
@@ -0,0 +1,28 @@
+-- EXECUTE IMMEDIATE is necessary to use @date in the query for EXTERNAL_QUERY,
+-- because it requires a string literal
+EXECUTE IMMEDIATE CONCAT(
+  """
+  CREATE OR REPLACE TABLE
+    waitlist_v1
+  PARTITION BY
+    DATE(updated_at)
+  AS
+  SELECT
+    IF(
+      _update.id IS NULL
+      OR waitlist_v1.updated_at > _update.updated_at,
+      waitlist_v1,
+      _update
+    ).*
+  FROM
+    EXTERNAL_QUERY(
+      "moz-fx-guardian-prod-bfc7.us.guardian-sql-prod",
+      "SELECT * FROM vpn_waitlist WHERE DATE(updated_at) = DATE '""",
+  FORMAT_DATE("%F", @date),
+  """';"
+    ) AS _update
+  FULL JOIN
```

Comment:
Why express this with this join against the existing table rather atomically replacing just the partition for the target date?

It seems most flexible to do a daily extract of the previous day into BQ, and then worry about determining the current state per ID in a separate step, whether materialized or a view.

Reply:
I think you are trying to think of this as a stream of updates, when it isn't? I am trying to mirror a cloud sql table. Instead of reading the whole table from cloud sql every day, this query aims to reduce impact on the prod cloud sql db by reading only the rows that changed.

I don't understand what benefit there is to materializing the result of the `EXTERNAL_QUERY` as a separate step. The result is only useful for updating rows to match the cloud sql table, and it could result in people querying the wrong table.